File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  include SessionsHelper
  
  def subscribed
    stream_form 'chat_channel'
    stream_for current_user.id
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    if data["content"]
      Chat.create!(
        user_id: data["current_user_id"].to_i,
        partner_id: data["partner_id"].to_i ,
        content: data["content"]
      )
      #画面を開いているのがチャット送信者だった場合
      ChatChannel.broadcast_to data["current_user_id"].to_i,
        content: data["content"],
        partner_id: data["partner_id"],
        isCurrent_user: true

      #画面を開いているのがチャット受信者だった場合
      ChatChannel.broadcast_to data["partner_id"].to_i,
        content: data["content"],
        partner_id: data["partner_id"],
        isCurrent_user: false
    end
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("ChatChannel", {
  connected: function() {
    // Called when the subscription is ready for use on the server
  },
  disconnected: function() {
    // Called when the subscription has been terminated by the server
  },
  received: function(data) {
    //画面を開いているのがチャット送信者だった場合
    if (data["isCurrent_user"]==true){
      content=`<div class='mycomment'><p>${data["content"]}</p></div>`;
    }
    //画面を開いているのがチャット受信者だった場合
    else{
      content=`<div class='fukidasi'><div class='faceicon'>
      <img src='/assets/profile.png' alt='管理人'></div>
      <div class='chatting'><div class='says'><p>${data["content"]}</p>
      </div></div></div>`;
    }
    $('#chats').append(content);
  },
  speak: function(content) {
    current_user_id=$("#current_user_id").val();
    partner_id=$("#partner_id").val();
    return this.perform('speak',{content: content, current_user_id: current_user_id, partner_id: partner_id});
  }
});
$(function(){
  $("#send").on("click",function(e){
    content=$("#content").val();
    App.chat.speak(content);
    $("#content").val(""); //フォームを空に
    e.preventDefault();
  });
});